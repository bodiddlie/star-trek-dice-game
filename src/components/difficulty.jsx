import React from 'react';
import { connect } from 'react-redux';

export const Difficulty = ({ difficulty }) =>
  <div>
    {difficulty}
  </div>;

const mapStateToProps = state => ({
  difficulty: state.difficulty,
});

export default connect(mapStateToProps)(Difficulty);
