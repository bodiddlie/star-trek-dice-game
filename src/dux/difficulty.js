export default function reducer(state = 0, action) {
  switch (action.type) {
    case SET_DIFFICULTY: {
      return action.difficulty;
    }
    default: {
      return state;
    }
  }
}

//----------------------
// ACTIONS
//----------------------
const SET_DIFFICULTY = '[Difficulty] Set Difficulty';

export function setDifficulty(difficulty) {
  return { type: SET_DIFFICULTY, difficulty };
}
