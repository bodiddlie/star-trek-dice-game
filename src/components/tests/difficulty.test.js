import React from 'react';
import { shallow } from 'enzyme';

import { Difficulty } from '../difficulty';

describe('<Difficulty />', () => {
  it('renders the chosen difficulty', () => {
    const wrapper = shallow(<Difficulty difficulty={2} />);
    expect(wrapper).toMatchSnapshot();
  });
});
