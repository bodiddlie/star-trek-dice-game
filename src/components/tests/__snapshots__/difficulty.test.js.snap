// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Difficulty /> renders the chosen difficulty 1`] = `
<div>
  2
</div>
`;
